pandas/core/indexes/base.py

```diff
@@ -6949,7 +6950,15 @@ def drop(
             if errors != "ignore":
                 raise KeyError(f"{labels[mask].tolist()} not found in axis")
             indexer = indexer[~mask]
-        return self.delete(indexer)
+        new_index = self.delete(indexer)
+
+        # check if we need to set the freq attribute
+        from pandas import DatetimeIndex
+
+        if isinstance(self, DatetimeIndex) and isinstance(new_index, DatetimeIndex):
+            new_index.freq = to_offset(new_index.inferred_freq)
```

Comment:
inferred_freq can be expensive.  _might_ be cheaper to invert `indexer` and call `take`, which has another PR to preserve freq which we can optimize